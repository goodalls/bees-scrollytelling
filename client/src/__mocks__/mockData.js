module.exports = {
  appDefaultState: {
    scrollPosition: 750,
    questionsEnabled: true,
    currentUserId: null,
    displayGraphCover: true
  },
  event: {
    target: {
      name: 'Jeff',
      value: 'websocket'
    }
  },
  beeFactsPost: {
    body: '{"users_id":2,"user_answer":"","question":"bee species"}',
    headers: { 'Content-type': 'application/json' },
    method: 'POST'
  },
  headerDefaultState: {
    concern: '',
    age: '',
    location: {},
    answered: false
  },
  impactDefaultState: {
    percent: 0,
    unit: 'million',
    monetary: '',
    economy: '',
    answeredeconomy: false,
    answeredpercent: false
  },
  impactFactsPost: {
    body: '{"users_id":2,"question":"are plants great"}',
    headers: { 'Content-type': 'application/json' },
    method: 'POST'
  },
  problemsPostBody: {
    method: 'POST',
    headers: {
      'Content-type': 'application/json'
    },
    body: JSON.stringify({
      users_id: 2,
      user_answer: 30,
      question: 'What percent of bee colonies do beekeepers lose every year?'
    })
  },
  rawStatistics: [
    {
      id: 48,
      question:
        'What percentage of your diet do you think is provided by bees?',
      user_answer: '35',
      users_id: 50
    },
    {
      id: 49,
      question: 'How much do bees contribute to the economy, anually?',
      user_answer: '400_million',
      users_id: 50
    },
    { id: 50, question: 'bee species', user_answer: '200', users_id: 50 },
    {
      id: 51,
      question: 'What percent of bee colonies do beekeepers lose every year?',
      user_answer: '40',
      users_id: 50
    },
    {
      id: 52,
      question:
        'What percentage of your diet do you think is provided by bees?',
      user_answer: '80',
      users_id: 51
    },
    {
      id: 53,
      question: 'How much do bees contribute to the economy, anually?',
      user_answer: '400_billion',
      users_id: 51
    },
    { id: 54, question: 'bee species', user_answer: '100', users_id: 51 },
    {
      id: 55,
      question: 'What percent of bee colonies do beekeepers lose every year?',
      user_answer: '25',
      users_id: 51
    }
  ],
  cleanedStatistics: {
    'How much do bees contribute to the economy, anually?': {
      count: { '400_million': 1, '400_billion': 1 },
      sum: 400400000000
    },
    'What percent of bee colonies do beekeepers lose every year?': {
      count: { 25: 1, 40: 1 },
      sum: 65
    },
    'What percentage of your diet do you think is provided by bees?': {
      count: { 35: 1, 80: 1 },
      sum: 115
    },
    'bee species': { count: { 100: 1, 200: 1 }, sum: 300 }
  },
  cleanedAverageData: {
    'How much do bees contribute to the economy, anually?': {
      count: { '400_million': 1, '400_billion': 1 },
      sum: 400400000000,
      average: 200200000000
    },
    'What percent of bee colonies do beekeepers lose every year?': {
      count: { 25: 1, 40: 1 },
      sum: 65,
      average: 32.5
    },
    'What percentage of your diet do you think is provided by bees?': {
      count: { 35: 1, 80: 1 },
      sum: 115,
      average: 57.5
    },
    'bee species': { count: { 100: 1, 200: 1 }, sum: 300, average: 150 }
  },
  pieChartDefaultState: {
    class: '',
    degree: 0,
    percent: 10
  },
  percentEvent: {
    target: {
      name: 'percent',
      value: 55
    }
  }
};
