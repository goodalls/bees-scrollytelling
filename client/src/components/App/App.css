/*color variables*/
*, :after, :before {
  box-sizing: border-box; }

body {
  display: flex;
  justify-content: center;
  margin: 0 10%;
  align-items: center;
  font-weight: 300;
  font-family: 'Open Sans Condensed', sans-serif; }

form {
  position: relative;
  align-items: center;
  display: flex;
  flex-direction: column;
  width: 100%; }
  form label {
    align-self: left;
    margin: 3px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: flex-end; }
  form h2 {
    align-self: left;
    font-size: 5em;
    margin: 20px 0; }

input:focus {
  outline: none; }

input[type="text"] {
  margin: 10px 0 50px;
  font-size: 1.5em;
  height: 50px; }

button {
  height: 62px;
  border: none;
  background-color: #ffffff;
  color: #60CA9B;
  font-size: 1.5em;
  margin: 30px auto 10px;
  width: 162px;
  border: 3px solid #60CA9B;
  border-radius: 51px;
  transition: background-color 0.5s ease;
  outline: none; }
  button:hover {
    cursor: pointer;
    background-color: #60CA9B;
    color: white; }

.question-descr {
  font-size: 2em; }

#displayAll {
  display: block; }

#displayNone {
  display: none; }

.progress-container {
  transform: rotate(90deg);
  position: fixed;
  top: calc(96% - 360px);
  left: calc(2% - 360px);
  height: 30px;
  width: 100vh; }

.progress-bar {
  height: 30px;
  background-image: url("../../assets/bee3.png");
  background-repeat: no-repeat;
  background-position: right top;
  background-size: 30px 30px; }

.arrow-cont {
  height: 150px; }

.hover-arrow {
  animation: MoveUpDown 2s linear infinite;
  position: absolute;
  bottom: 0;
  width: 100px; }

@keyframes MoveUpDown {
  0%, 100% {
    bottom: 0; }
  50% {
    bottom: 50px; } }
