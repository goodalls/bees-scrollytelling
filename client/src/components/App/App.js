import React, { Component } from 'react';
import SwitchButton from '../SwitchButton/SwitchButton';
import Statistics from '../Statistics/Statistics';
import BeeFacts from '../BeeFacts/BeeFacts';
import Problems from '../Problems/Problems';
import Header from '../Header/Header';
import Impact from '../Impact/Impact';
import Help from '../Help/Help';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      scrollPosition: 750,
      questionsEnabled: true,
      currentUserId: null,
      displayGraphCover: true
    };
  }

  componentDidMount = () => {
    window.addEventListener('scroll', this.stopScroll);
  };

  stopScroll = () => {
    this.manageGraphDisplay();
    this.trackBeeScroll();
    if (
      this.state.questionsEnabled &&
      window.scrollY > this.state.scrollPosition
    ) {
      window.scrollTo(0, this.state.scrollPosition);
    }
  };

  manageGraphDisplay = () => {
    if (window.scrollY > 5010 && this.state.displayGraphCover) {
      console.log('changed')
      this.setState({ displayGraphCover: false })
    } else if (!this.state.displayGraphCover && window.scrollY < 4561) {
      this.setState({ displayGraphCover: true })
    }
  };

  startScroll = (e, scrollPosition) => {
    e.preventDefault();
    this.setState({ scrollPosition });
  };

  saveUser = id => {
    this.setState({
      currentUserId: id
    });
  };

  toggleQuestionsEnabled = () => {
    this.setState({
      questionsEnabled: !this.state.questionsEnabled
    });
  };

  

  trackBeeScroll = () => {
    const winScroll =
      document.body.scrollTop || document.documentElement.scrollTop;
    const height =
      document.documentElement.scrollHeight -
      document.documentElement.clientHeight;
    const scrolled = winScroll / height * 100;
    document.getElementById('bee-bar').style.width = scrolled + '%';
  };

  render() {
    return (
      <div className="App" onScroll={() => console.log('hi')}>
        <div className="progress-container">
          <div className="progress-bar" id="bee-bar" />
        </div>
        <SwitchButton toggleQuestionsEnabled={this.toggleQuestionsEnabled} />
        <Header
          startScroll={this.startScroll}
          questionsEnabled={this.state.questionsEnabled}
          saveUser={this.saveUser}
        />
        <Impact
          startScroll={this.startScroll}
          questionsEnabled={this.state.questionsEnabled}
          currentUserId={this.state.currentUserId}
          displayGraphCover={this.state.displayGraphCover}
        />
        <BeeFacts
          questionsEnabled={this.state.questionsEnabled}
          startScroll={this.startScroll}
          currentUserId={this.state.currentUserId}
        />
        <Problems
          questionsEnabled={this.state.questionsEnabled}
          startScroll={this.startScroll}
          currentUserId={this.state.currentUserId}
        />
        <Help questionsEnabled={this.state.questionsEnabled} />
        <Statistics userId={this.state.currentUserId} />
      </div>
    );
  }
}

export default App;
