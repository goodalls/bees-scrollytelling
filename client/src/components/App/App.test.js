import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import mockData from '../../__mocks__/mockData';

describe('App', () => {
  let wrapper;
  global.window = window;
  window.addEventListener = jest.fn();
  window.scrollTo = jest.fn();
  // let document;
  // global.document = document;
  // document = {
  //   documentElement: {
  //     clientHeight: 600
  //   },
  //   querySelector: jest.fn().mockImplementation(() => {
  //     return {
  //       style: {
  //         width: 0
  //       }
  //     }
  //   })
  // }
  // console.log(document)
  // // global.document.documentElement = document.body.clientHeight = 600;
  

  // // Object.defineProperty(document, 'documentElement', { clientHeight: 600 })
  beforeEach(() => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true });
    window.addEventListener.mockClear();
    window.scrollTo.mockClear();
    
  });

  it('should match the Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has default state', () => {
    expect(wrapper.instance().state).toEqual(mockData.appDefaultState);
  });

  describe('componentDidMount', () => {
    it('calls addEventListener on window', () => {
      
      expect(window.addEventListener).not.toHaveBeenCalled();

      wrapper.instance().componentDidMount();
   
      expect(window.addEventListener).toHaveBeenCalled();
    });
  });

  describe('stopScroll', () => {
    it('calls scrollTo on window when questions are enabled and scroll position is greater that current limit', () => {
      wrapper.instance().trackBeeScroll = jest.fn();
      expect(window.scrollTo).not.toHaveBeenCalled();
      window.scrollY = 800;

      wrapper.instance().stopScroll();

      expect(window.scrollTo).toHaveBeenCalledWith(0, 750);
    });
  });

  describe('startScroll', () => {
    let event = {
      preventDefault: jest.fn()
    }

    it('calls preventDefault on event passed in', () => {
      expect(event.preventDefault).not.toHaveBeenCalled();

      wrapper.instance().startScroll(event, 800);

      expect(event.preventDefault).toHaveBeenCalled();
    });

    it('sets state with scrollPosition passed in', () => {
      wrapper.instance().startScroll(event, 800);
      expect(wrapper.instance().state.scrollPosition).toEqual(800);
    });
  });

  describe('saveUser', () => {
    it('sets state with user id passed in', () => {
      wrapper.instance().saveUser(9);
      expect(wrapper.instance().state.currentUserId).toEqual(9);
    });
  });

  describe('toggleQuestionsEnabled', () => {
    it('sets questionsEnabled to opposite of current state', () => {
      wrapper.instance().toggleQuestionsEnabled();

      expect(wrapper.instance().state.questionsEnabled).toEqual(false);

      wrapper.instance().toggleQuestionsEnabled();

      expect(wrapper.instance().state.questionsEnabled).toEqual(true);
    });
  });
});
