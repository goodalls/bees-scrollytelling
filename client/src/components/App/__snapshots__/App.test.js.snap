// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="progress-container"
>
          <div
                    className="progress-bar"
                    id="bee-bar"
          />
</div>,
        <SwitchButton
          toggleQuestionsEnabled={[Function]}
/>,
        <Header
          questionsEnabled={true}
          saveUser={[Function]}
          startScroll={[Function]}
/>,
        <Impact
          currentUserId={null}
          displayGraphCover={true}
          questionsEnabled={true}
          startScroll={[Function]}
/>,
        <BeeFacts
          currentUserId={null}
          questionsEnabled={true}
          startScroll={[Function]}
/>,
        <Problems
          currentUserId={null}
          questionsEnabled={true}
          startScroll={[Function]}
/>,
        <Help
          questionsEnabled={true}
/>,
        <Statistics
          userId={null}
/>,
      ],
      "className": "App",
      "onScroll": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="progress-bar"
            id="bee-bar"
/>,
          "className": "progress-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "progress-bar",
            "id": "bee-bar",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "toggleQuestionsEnabled": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "questionsEnabled": true,
          "saveUser": [Function],
          "startScroll": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currentUserId": null,
          "displayGraphCover": true,
          "questionsEnabled": true,
          "startScroll": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currentUserId": null,
          "questionsEnabled": true,
          "startScroll": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currentUserId": null,
          "questionsEnabled": true,
          "startScroll": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "questionsEnabled": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "userId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="progress-container"
>
            <div
                        className="progress-bar"
                        id="bee-bar"
            />
</div>,
          <SwitchButton
            toggleQuestionsEnabled={[Function]}
/>,
          <Header
            questionsEnabled={true}
            saveUser={[Function]}
            startScroll={[Function]}
/>,
          <Impact
            currentUserId={null}
            displayGraphCover={true}
            questionsEnabled={true}
            startScroll={[Function]}
/>,
          <BeeFacts
            currentUserId={null}
            questionsEnabled={true}
            startScroll={[Function]}
/>,
          <Problems
            currentUserId={null}
            questionsEnabled={true}
            startScroll={[Function]}
/>,
          <Help
            questionsEnabled={true}
/>,
          <Statistics
            userId={null}
/>,
        ],
        "className": "App",
        "onScroll": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="progress-bar"
              id="bee-bar"
/>,
            "className": "progress-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "progress-bar",
              "id": "bee-bar",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "toggleQuestionsEnabled": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "questionsEnabled": true,
            "saveUser": [Function],
            "startScroll": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentUserId": null,
            "displayGraphCover": true,
            "questionsEnabled": true,
            "startScroll": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentUserId": null,
            "questionsEnabled": true,
            "startScroll": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentUserId": null,
            "questionsEnabled": true,
            "startScroll": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionsEnabled": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "userId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
